lib/IRGen/GenEnum.cpp

```diff
@@ -3549,10 +3565,10 @@ namespace {
             elt.decl, IGM.getSILModule(), IGM.getMaximalTypeExpansionContext());
 
         nonEmptyCases.push_back(
-            elt.ti->buildTypeLayoutEntry(IGM, eltPayloadType));
+            elt.ti->buildTypeLayoutEntry(IGM, eltPayloadType, useStructLayouts));
       }
-      return IGM.typeLayoutCache.getOrCreateEnumEntry(emptyCases,
-                                                      nonEmptyCases);
+      return IGM.typeLayoutCache.getOrCreateEnumEntry(emptyCases, nonEmptyCases,
```

Comment:
Looking at this code, I am worried about the following scenario. I think there is a problem here.

- We have enabled `useStructLayouts`
- We generate code for one of the value witness functions that don't use the byte code string e.g. assignWithCopy
- The Enum is loadable
- Before this change we  would have used the `TypeInfoBasedEntry` method
- Now we use the `EnumEntry` method
- Note the comment above "type layout entry code does not handle spare bits atm"; the code generation in type layout's `EnumEntryTypeLayout` for assign with copy does not do the right thing for a multipayload enum where payloads have spare bits because that is not implemented.

Reply:
You can fix that by unconditionally using the type info based emission in `GenValueWitnesses.cpp` is the enum is fixed size (`AllowFixedLayoutOptimizations && TIK >= Loadable`), I think

Reply:
I changed the mechanism used to enable layout strings, so we don't have to unconditionally enable struct layouts. If layout strings are enabled, we get the struct layout for the layout string generation only. The other witnesses will still use `TypeInfoBasedEntry` unless struct layouts are explicitly enabled.

Reply:
Okay, great.